### PR TITLE
Basenames: add nameExpires to the tokenMetadata JSON

### PR DESCRIPTION
**Pull request by kirkas**
image

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: apps/web/src/utils/usernames.ts [hunk 1]
```diff
@@ -583,6 +583,28 @@ export async function getBasenameOwner(username: BaseName) {
...
+
+export async function getBasenameNameExpires(username: BaseName) {
+  const chain = getChainForBasename(username);
+
+  try {
+    const client = getBasenamePublicClient(chain.id);
+    const nameExpires = await client.readContract(buildBasenameNameExpiresContract(username));
+
+    return nameExpires;
+  } catch (error) {}
```
Comment (JFrankfurt) [inline apps/web/src/utils/usernames.ts:605]: imo utilities like this should not silently disappear their own errors.

File: apps/web/src/utils/usernames.ts [hunk 2]
```diff
@@ -583,6 +583,28 @@ export async function getBasenameOwner(username: BaseName) {
   } catch (error) {}
 }
 
+export function buildBasenameNameExpiresContract(username: BaseName): ContractFunctionParameters {
```
Comment (JFrankfurt) [inline apps/web/src/utils/usernames.ts:586]: nit: it looks like this isn't used elsewhere--should prob inline into getBasenameNameExpires, or at least not export

File: apps/web/src/utils/usernames.ts (hunk 1 above)
Comment (kirkas) [reply apps/web/src/utils/usernames.ts:605]: I think that's fair but we do that every where now, so maybe a larger task?

**Review by zencephalon — APPROVED** [review]

**Pull request merged**